main.cpp

```diff
@@ -270,101 +219,165 @@ struct HeapA
     A* a = nullptr;
 };
 
+template<typename T>
+struct Numeric
+{
+    using Type = T;
+    explicit Numeric(Type val) : value(std::make_unique<Type>(val)) {}
+    ~Numeric()
+    {
+        value.reset(nullptr);
+    }
```

Comment:
This destructor is unnecessary.  
`unique_ptr` automatically destroys the owned object when it goes out of scope, which is what happens when the `Numeric<>` class is destroyed.
- [ ] revise